src/Tickets/Emails/Email_Template.php

```diff
@@ -0,0 +1,193 @@
+	 *
+	 * @since TBD
+	 *
+	 * @param Array $data
+	 * 
+	 * @return void
+	 */
+	public function set_data( Array $data ) {
+		$this->context_data = $data;
+	}
+
+	/**
+	 * Returns the template context array and creates sample data if preview.
+	 *
+	 * @since TBD
+	 *
+	 * @return Array Template context array.
+	 */
+	public function get_context() {
+		$context = [
+			'preview'                => $this->preview,
+			'header_image_url'       => tribe_get_option( Settings::$option_header_image_url, '' ),
+			'header_image_alignment' => tribe_get_option( Settings::$option_header_image_alignment, 'left' ),
+			'header_bg_color'        => tribe_get_option( Settings::$option_header_bg_color, '#ffffff' ),
+			'ticket_bg_color'        => tribe_get_option( Settings::$option_ticket_bg_color, '#ffffff' ),
+			'footer_content'         => tribe_get_option( Settings::$option_footer_content, '' ),
+			'footer_credit'          => tribe_get_option( Settings::$option_footer_credit, true ),
+		];
+		$context['header_text_color'] = $this->get_contrast_color( $context['header_bg_color'] );
+		$context['ticket_text_color'] = $this->get_contrast_color( $context['ticket_bg_color'] );
+
+		if ( $this->preview ) {
+			$current_user = wp_get_current_user();
+			$this->context_data = [
+				'recipient_first_name' => $current_user->first_name,
+				'recipient_last_name'  => $current_user->last_name,
+				'date_string'          => esc_html__( 'September 22 @ 7:00 pm - 11:00 pm', 'event-tickets' ),
+				'qr_url'               => esc_url( plugins_url( '/event-tickets/src/resources/images/example-qr.png' ) ),
+				'ticket_name'          => esc_html__( 'General Admission', 'event-tickets' ),
+				'ticket_id'            => '17e4a14cec',
+				'event_title'          => esc_html__( 'Rebirth Brass Band', 'event-tickets' ),
+				'event_image_url'      => esc_url( plugins_url( '/event-tickets/src/resources/images/example-event-image.png' ) ),
+				'event_venue'          => [
+					'name'     => esc_html__( 'Saturn', 'event-tickets' ),
+					'address1' => esc_html__( '200 41st Street South', 'event-tickets' ),
+					'address2' => esc_html__( 'Birmingham, AL, 35222', 'event-tickets' ),
+					'phone'    => esc_html__( '(987) 654-3210', 'event-tickets' ),
+					'website'    => esc_url( get_site_url() ),
+				],
+				'event_description' => '<h4>Additional Information</h4><p>Age Restriction: 18+<br>Door Time: 8:00PM<br>Event Time: 9:00PM</p>',
+			];
+		}
+		error_log(var_export($this->context_data, true));
+		return array_merge( $context, $this->context_data );
+		// return apply_filters( 'tec_tickets_emails_email_template_context', $context );
+	}
+
+	/**
+	 * Returns contrasting color (light or dark) based on input color.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $hexColor 6-character hexidecimal color code, including hash.
+	 * 
+	 * @return string Contrasting 6-character hexidecimal color code.
+	 */
+	private function get_contrast_color( $hexColor ) {
```

Comment:
We can possibly move this one to a "color" helper class in common. @bordoni thoughts?

Also, add the return type.

Reply:
The code of this method also needs the proper spacing, dots after comments, etc.

Reply:
Ah. Yeah. I copied a lot of the code and didn't update the comments to meet our standards. I do like the idea of moving it to the color helper class. Any word from @bordoni on this, yet?

Reply:
Spoke with @bordoni . He agreed on moving the method to the `Tribe__Utils__Color` class. That PR is here: https://github.com/the-events-calendar/tribe-common/pull/1855